service/lxd.go

```diff
@@ -481,6 +481,47 @@ func (s LXDService) GetCephInterfaces(ctx context.Context, bootstrap bool, peers
 	return candidates, nil
 }
 
+// ValidateCephInterfaces validates the given interfaces map against the given Ceph network subnet
+// and returns a map of peer name to interfaces that are in the subnet.
+func (s *LXDService) ValidateCephInterfaces(cephNetworkSubnetStr string, interfacesMap map[string][]CephDedicatedInterface) (map[string][][]string, error) {
+	_, subnet, err := net.ParseCIDR(cephNetworkSubnetStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid CIDR subnet: %v", err)
+	}
+
+	ones, bits := subnet.Mask.Size()
+	if bits-ones == 0 {
+		return nil, fmt.Errorf("invalid Ceph network subnet (must be a subnet with at least one address)")
```

Comment:
What is the convention in microcloud for capitalisation of errors? In LXD they are all capitalised.

Also, if `bits-ones == 0`, wouldn't there be exactly one address? (e.g. `10.123.123.1/32`). Should we instead have `Invalid Ceph network subnet (must have more than one address)`?

Reply:
fixed